ext/standard/math.c

```diff
@@ -165,56 +121,30 @@ static inline double php_round_helper(double value, int mode) {
 
 /* {{{ _php_math_round */
 /*
- * Rounds a number to a certain number of decimal places in a certain rounding
- * mode. For the specifics of the algorithm, see http://wiki.php.net/rfc/rounding
+ * Rounds a number to a certain number of decimal places in a certain rounding mode.
+ * If you "HALF UP" a value like 0.285 (0.28499999999999998), it will be rounded to 0.28.
```

Comment:
`0.28499999999999998` is theoretical meaningless value, the stored accuracy is lower.

`0.285` is simply `0.285`, the rounding must match double to string representation /w `precision` set to `-1`.

Reply:
please see #12268

To be honest, I'm starting to feel like it's impossible for me to reconcile the divided opinions, and I'm starting to think about posting about this on internals.

Reply:
see https://github.com/php/php-src/blob/php-8.2.10/Zend/zend_strtod.c#L3704 for double -> string conversion code /w mode=0, not sure if rounding thru string conversion will be fast enough, but in the end, all roundings must equal when does this way